### PR TITLE
Find one and...

### PR DESCRIPTION
**Pull request by tkaye407**

### CODE REVIEW

File: packages/core/services/mongodb-remote/src/RemoteFindOneAndDeleteOptions.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  
+    /**
+     * Optional: Specifies the query sort order. Sort documents specify one or more fields to 
+     * sort on  where the value of each field indicates whether MongoDB should sort it in 
+     * ascending (1) or descending (0) order. 
+     * The sort order determines which document collection.findOneAndDelete() affects.
+     */
+    readonly sort?: object;
+  }
+
```
Comment (adamchel) [inline packages/core/services/mongodb-remote/src/RemoteFindOneAndDeleteOptions.ts]: [nit] remove extra newline from file

File: packages/core/services/mongodb-remote/src/RemoteFindOneAndUpdateOptions.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    readonly upsert?: boolean;
+
+    /* 
+     * Optional. Default: false. 
+     * A boolean that, if true, indicates that the action should return 
+     * the document in its updated form instead of its original, pre-update form.
+     */
+    readonly returnNewDocument?: boolean;
+  }
+
```
Comment (adamchel) [inline packages/core/services/mongodb-remote/src/RemoteFindOneAndUpdateOptions.ts]: [nit] remove extra newline

**Comment by adamchel** [comment]
Though it looks like evergreen is failing due to some `null` stuff?

File: packages/browser/services/mongodb-remote/__tests__/RemoteMongoClientIntTests.ts [hunk 3]
```diff
@@ -251,6 +252,265 @@ describe("RemoteMongoClient", () => {
...
   });
 
+  it("should find one and update", async () => {
+    const coll = getTestColl();
+
+    // Collection should start out empty
+    // This also tests the null return format
+    let result = await coll.findOneAndUpdate({}, {});
+    expect(result).toEqual({});
+    // expect(result).toBeNull();
```
[the remark this answers is not in the document]
Comment (tkaye407) [reply packages/browser/services/mongodb-remote/__tests__/RemoteMongoClientIntTests.ts]: Ohh, sorry waiting on a change from server because the comment is supposed to be correct

File: packages/browser/services/mongodb-remote/__tests__/RemoteMongoClientIntTests.ts [hunk 4]
```diff
@@ -251,6 +252,265 @@ describe("RemoteMongoClient", () => {
...
+    expect(withoutId(result)).toEqual({hello: "hellothere", num: 4});
+
+    // Test null behavior again with filter that should not match any docs
+    result = await coll.findOneAndUpdate(
+      {hello: "hellotherethisisnotakey"}, 
+      {$inc: {num: 1}}, 
+      {returnNewDocument: true}
+    )
+    expect(result).toEqual({});
+    // expect(result).toBeNull();
```
[the remark this answers is not in the document]
Comment (tkaye407) [reply packages/browser/services/mongodb-remote/__tests__/RemoteMongoClientIntTests.ts]: Same as above

File: packages/core/services/mongodb-remote/src/RemoteFindOneAndDeleteOptions.ts (hunk 1 above)
Comment (tkaye407) [reply packages/core/services/mongodb-remote/src/RemoteFindOneAndDeleteOptions.ts]: I thought we always put an extra line at the end of the file?

File: packages/browser/services/mongodb-remote/__tests__/RemoteMongoClientIntTests.ts [hunk 5]
```diff
@@ -251,6 +250,261 @@ describe("RemoteMongoClient", () => {
...
+    // Collection should start out empty
+    // This also tests the null return format
+    let result = await coll.findOneAndUpdate({}, {});
+    expect(result).toBeNull();
+
+    // Insert Sample Document
+    const doc1 = { hello: "world", num: 2 };
+    await coll.insertOne(doc1);
+
+    // Simple call to findOneAndUpdate() were we get the previous document back
```
Comment (tkaye407) [inline packages/browser/services/mongodb-remote/__tests__/RemoteMongoClientIntTests.ts]: spelling

File: packages/react-native/services/mongodb-remote/__tests__/RemoteMongoClientIntTests.ts [hunk 6]
```diff
@@ -245,6 +245,261 @@ describe("RemoteMongoClient", () => {
...
+    
+    // Test sort and project
+    result = await coll.findOneAndDelete({}, {projection: {hello: 1, _id: 0}, sort: {num: -1}});
+    expect(result).toEqual({hello: "world3"})
+    result = await coll.findOneAndDelete({}, {projection: {hello: 1, _id: 0}, sort: {num: 1}});
+    expect(result).toEqual({hello: "world1"})
+    
+    // Should properly fail given an illegal replacement doc with update operations
+    try {
+      await coll.findOneAndReplace({}, {sort: {$inc: {a: 1}}})
```
Comment (tkaye407) [inline packages/react-native/services/mongodb-remote/__tests__/RemoteMongoClientIntTests.ts]: This is wrong, and I dont think it will fail
